File: designPattern/Composite/Basic/src/File.java

public class File extends Entry{
    String name;
    int size;
    public File(String name, int size){
        this.name = name;
        this.size = size;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getSize() {
        return size;
    }

    @Override
    public void printLine(String prefix) {
        System.out.println(prefix+"/"+this);
    }

    @Override
    public Entry add(Entry entry) {
        return null;
    }
}

File: designPattern/Composite/basicPython/File.py
from Entry import Entry


class File(Entry):
    def __init__(self, name, size):
        self.name = name
        self.size = size

    def getName(self):
        return self.name

    def getSize(self):
        return self.size

    def print_line(self, prefix):
        print(prefix+"/"+self.getInformation())